test/service/healthcheck_http_server_test.cc

```diff
@@ -0,0 +1,124 @@
+#include "src/service/healthcheck_http_server.h"
+
+#include <memory>
+
+#include "boost/asio/io_context.hpp"
+#include "boost/asio/ip/tcp.hpp"
+#include "boost/asio/spawn.hpp"
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+#include "src/common/http/http.h"
+#include "src/config/get_config.h"
+#include "src/filters/filter_chain.h"
+#include "test/filters/oidc/mocks.h"
+
+namespace authservice {
+namespace service {
+
+using testing::_;
+using testing::Return;
+using testing::ReturnRef;
+
+TEST(TestHealthCheckHttpServer, BasicFlowWithDanglingJwks) {
+  boost::asio::io_context io_context;
+
+  auto configuration = std::make_unique<config::FilterChain>();
+  auto chain =
+      std::make_unique<filters::FilterChainImpl>(io_context, *configuration, 1);
+
+  auto mock_resolver = std::make_shared<filters::oidc::MockJwksResolver>();
+  google::jwt_verify::JwksPtr dangling_jwks;
+  EXPECT_CALL(*mock_resolver, jwks())
+      .Times(2)
+      .WillRepeatedly(ReturnRef(dangling_jwks));
+
+  auto resolver_cache =
+      std::make_unique<filters::oidc::MockJwksResolverCache>();
+  EXPECT_CALL(*resolver_cache, getResolver())
+      .Times(2)
+      .WillRepeatedly(Return(mock_resolver));
+
+  chain->setJwksResolverCache(std::move(resolver_cache));
+  EXPECT_FALSE(chain->jwksActive());
+
+  std::vector<std::unique_ptr<filters::FilterChain>> chains;
+  chains.push_back(std::move(chain));
+
+  HealthcheckAsyncServer server(chains, "0.0.0.0", 33333);
+
+  auto http_ptr = common::http::ptr_t(new common::http::HttpImpl);
```

Comment:
It takes me a bit browsing to realize this is a shared_ptr. Is this the common typedef naming pattern we use in other places as well?

Also I think here unique_ptr would be sufficient right? And unique_ptr is preferred over shared_ptr when possible IIRC.